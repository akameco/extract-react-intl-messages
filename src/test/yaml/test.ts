import fs from 'fs'
import path from 'path'
// eslint-disable-next-line import/no-extraneous-dependencies
import tempy from 'tempy'
// eslint-disable-next-line import/no-extraneous-dependencies
import tempWrite from 'temp-write'
import yaml from 'js-yaml'
import m from '../..'

const fixturesPath = 'src/test/fixtures/default/**/*.js'

const yamlLoad = (tmp: string, file = '') =>
  yaml.safeLoad(fs.readFileSync(path.resolve(tmp, file), 'utf8'))

const defaultLocale = 'en'

test('export yaml', async () => {
  const tmp = tempy.directory()
  await m(['en', 'ja'], fixturesPath, tmp, { defaultLocale, format: 'yaml' })

  expect(yamlLoad(tmp, 'en.yml')).toMatchSnapshot()
  expect(yamlLoad(tmp, 'ja.yml')).toMatchSnapshot()
})

test('export yaml - flat', async () => {
  const tmp = tempy.directory()
  await m(['en', 'ja'], fixturesPath, tmp, {
    defaultLocale,
    format: 'yaml',
    flat: true
  })

  expect(yamlLoad(tmp, 'en.yml')).toMatchSnapshot()
  expect(yamlLoad(tmp, 'ja.yml')).toMatchSnapshot()
})

test('exsit yaml', async () => {
  const x = { a: { hello: 'hello2' } }

  const tmpEn = tempWrite.sync(yaml.safeDump(x), 'en.yml')
  await m(['en'], fixturesPath, path.dirname(tmpEn), {
    defaultLocale,
    format: 'yaml'
  })
  expect(yaml.safeLoad(fs.readFileSync(tmpEn, 'utf8'))).toMatchSnapshot()

  const tmpJa = tempWrite.sync(yaml.safeDump(x), 'ja.yml')
  await m(['ja'], fixturesPath, path.dirname(tmpJa), {
    defaultLocale,
    format: 'yaml'
  })

  expect(yaml.safeLoad(fs.readFileSync(tmpJa, 'utf8'))).toMatchSnapshot()
})
