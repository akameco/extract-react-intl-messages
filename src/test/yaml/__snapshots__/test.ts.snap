// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`export yaml - flat 1`] = `
Object {
  "a.hello": "hello",
  "a.world": "world",
  "b.hello": "hello",
  "b.world": "world",
}
`;

exports[`export yaml - flat 2`] = `
Object {
  "a.hello": "",
  "a.world": "",
  "b.hello": "",
  "b.world": "",
}
`;

exports[`export yaml 1`] = `
Object {
  "a": Object {
    "hello": "hello",
    "world": "world",
  },
  "b": Object {
    "hello": "hello",
    "world": "world",
  },
}
`;

exports[`export yaml 2`] = `
Object {
  "a": Object {
    "hello": "",
    "world": "",
  },
  "b": Object {
    "hello": "",
    "world": "",
  },
}
`;

exports[`exsit yaml 1`] = `
Object {
  "a": Object {
    "hello": "hello",
    "world": "world",
  },
  "b": Object {
    "hello": "hello",
    "world": "world",
  },
}
`;

exports[`exsit yaml 2`] = `
Object {
  "a": Object {
    "hello": "hello2",
    "world": "",
  },
  "b": Object {
    "hello": "",
    "world": "",
  },
}
`;
